### PR TITLE
HRCPP-491 fixed mem leak

### PR DESCRIPTION
**Pull request by rigazilla**
This fixes a memory leak between C++ and C# layers

### CODE REVIEW

**Comment by rigazilla** [comment]
Hey @alanfx  can you have a look at this, it on the swig part. It's a quite annoying bug if user deals with large cache value. Thanks!

File: swig/CMakeLists.txt [hunk 1]
```diff
@@ -24,9 +24,9 @@ else (WIN32)
 endif (WIN32)
 
 if (CMAKE_MAJOR_VERSION LESS 3)
-  set_source_files_properties(${HOTROD_SWIG_MODULE_SOURCE} PROPERTIES SWIG_FLAGS "-v;-namespace;Infinispan.HotRod.SWIGGen;-dllimport;${HOTROD_WRAP_LIB}")
+  set_source_files_properties(${HOTROD_SWIG_MODULE_SOURCE} PROPERTIES SWIG_FLAGS "-DSWIG2_CSHARP;-v;-namespace;Infinispan.HotRod.SWIGGen;-dllimport;${HOTROD_WRAP_LIB}")
 else (CMAKE_MAJOR_VERSION LESS 3)
-  set_source_files_properties(${HOTROD_SWIG_MODULE_SOURCE} PROPERTIES SWIG_FLAGS "-v;-namespace;Infinispan.HotRod.SWIGGen;")
+	set_source_files_properties(${HOTROD_SWIG_MODULE_SOURCE} PROPERTIES SWIG_FLAGS "-DSWIG2_CSHARP;-v;-namespace;Infinispan.HotRod.SWIGGen;")
```
Comment (alanfx) [inline swig/CMakeLists.txt]: Indentation seems wrong here?

Comment (rigazilla) [reply swig/CMakeLists.txt]: vim on win makes me crazy :) should be ok now @alanfx

Comment (alanfx) [reply swig/CMakeLists.txt]: That would drive me crazy too, @rigazilla! Looks good now.

**Comment by alanfx** [comment]
Merging

**Pull request merged**